pkg/types/types.go

```diff
@@ -6,19 +6,22 @@ import (
 )
 
 type Plan struct {
-	Orchestration string                   `yaml:"orchestration"`
-	Secrets       []KeyValueNamespaceTuple `yaml:"secrets"`
-	RootDomain    string                   `yaml:"root_domain"`
-	Registry      string                   `yaml:"registry"`
-	CustomersURL  string                   `yaml:"customers_url"`
-	Github        Github                   `yaml:"github"`
-	TLS           bool                     `yaml:"tls"`
-	OAuth         OAuth                    `yaml:"oauth"`
-	S3            S3                       `yaml:"s3"`
-	EnableOAuth   bool                     `yaml:"enable_oauth"`
-	TLSConfig     TLSConfig                `yaml:"tls_config"`
-	Slack         Slack                    `yaml:"slack"`
-	Ingress       string                   `yaml:"ingress"`
+	Orchestration        string                   `yaml:"orchestration"`
+	Secrets              []KeyValueNamespaceTuple `yaml:"secrets"`
+	RootDomain           string                   `yaml:"root_domain"`
+	Registry             string                   `yaml:"registry"`
+	CustomersURL         string                   `yaml:"customers_url"`
+	Github               Github                   `yaml:"github"`
+	Gitlab               Gitlab                   `yaml:"gitlab"`
+	TLS                  bool                     `yaml:"tls"`
+	OAuth                OAuth                    `yaml:"oauth"`
+	S3                   S3                       `yaml:"s3"`
+	EnableOAuth          bool                     `yaml:"enable_oauth"`
+	TLSConfig            TLSConfig                `yaml:"tls_config"`
+	Slack                Slack                    `yaml:"slack"`
+	Ingress              string                   `yaml:"ingress"`
+	OAuthProvider        string                   `yaml:"oauth_provider"`
```

Comment:
I feel this should all be a sub-element of the OAuth property, do you agree?

Reply:
Yes, it's more clear